src/libraries/System.Speech/src/Internal/Synthesis/AudioBase.cs

```diff
@@ -121,7 +121,22 @@ internal void PlayWaveFile(AudioData audio)
                     try
                     {
                         byte[] data = new byte[(int)audio._stream.Length];
-                        audio._stream.Read(data, 0, data.Length);
+
+#if NET7_0_OR_GREATER
+                        audio._stream.ReadExactly(data);
+#else
+                        int totalRead = 0;
+                        while (totalRead < audio._stream.Length)
+                        {
+                            int bytesRead = audio._stream.Read(data, totalRead, audio._stream.Length - totalRead);
```

Comment:
NIT:
```suggestion
                        while (totalRead < data.Length)
                        {
                            int bytesRead = audio._stream.Read(data, totalRead, data.Length - totalRead);
```

Reply:
Thanks, I will use `data.Length` here, but I am not sure why this does not trigger CS1503 as well.